GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/GVREmitter.java

```diff
@@ -0,0 +1,350 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.particlesystem;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRDrawFrameListener;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRTexture;
+import org.gearvrf.particlesystem.Particles;
+import org.gearvrf.utility.Log;
+import org.joml.Vector3f;
+import org.joml.Vector4f;
+import java.util.ArrayList;
+
+
+/**
+ * The base emitter class from which shape-specific emitter classes inherit.
+ * This class is used to set up the the behaviour of the particle system in general.
+ * Also, this is used to set the specific particle properties.
+ *
+ * Every emitter has a set of scene objects as its children. These scene
+ * objects each have a mesh attached to them, with the number of vertices in the mesh
+ * being equal to the emit rate of the emitter, i.e., every second a new scene object with a mesh
+ * having an emitRate number of vertices is added to the emitter as a child.
+ * These vertices act as the particles of the system. Consequently, all those
+ * children objects which have exceeded their age limits are deleted every second.
+ *
+ */
+
+public class GVREmitter extends GVRSceneObject {
+
+    private int MAX_EMIT_RATE = 500;
+
+    protected int mEmitRate = 300;
+    protected boolean mEnableEmitter = true;
+    protected GVRContext mGVRContext = null;
+
+
+    //list of active scene objects containing particle meshes,
+    // and their corresponding spawn times.
+    protected ArrayList<GVRSceneObject> mActiveParticleMeshes = null;
+    protected ArrayList<Float> mParticleMeshSpawnTimes;
+
+
+    //particle properties
+    protected float mMaxAge = 1.5f;
+    protected float mParticleSize = 50.0f;
+    protected float minVelocity = 1.5f;
+    protected float maxVelocity = 4.5f;
+    protected Vector3f mEnvironmentAcceleration ;
+    private Vector4f mColor;
+    private float mParticleSizeRate = 0.0f;
+    private boolean mFadeWithAge = false;
+    private GVRTexture mParticleTexture;
+
+    //attributes of the particles generated every second.
+    protected float[] mParticlePositions;
+    protected float[] mParticleVelocities;
+    protected float [] mParticleGenTimes;
+
+    protected boolean burstMode = false;
+    private boolean executeOnce = true;
+
+    //values corresponding to the particles which degine the bounding volume of the system.
+    private float[] particleBoundingVolume;
+    private float[] BVSpawnTimes;
+    private float[] BVVelocities;
+
+    private float currTime = 0;
+
+    public GVREmitter(GVRContext gvrContext)
+    {
+        super(gvrContext);
+        mGVRContext = gvrContext;
+//        mGVRContext.registerDrawFrameListener(this);
```

Comment:
Please don't check in commented-out code without a good reason (good reason shall be explained in a detailed comment next to commented-out line)